src/Facebook/InstantArticles/Transformer/Transformer.php

```diff
@@ -8,48 +8,40 @@
  */
 namespace Facebook\InstantArticles\Transformer;
 
+use Facebook\InstantArticles\Transformer\Warnings\TransformerWarning;
 use Facebook\InstantArticles\Transformer\Warnings\UnrecognizedElement;
 use Facebook\InstantArticles\Transformer\Rules\Rule;
 use Facebook\InstantArticles\Elements\InstantArticle;
+use Facebook\InstantArticles\Elements\Element;
 use Facebook\InstantArticles\Validators\Type;
 use Facebook\InstantArticles\Validators\InstantArticleValidator;
 
 class Transformer
 {
     /**
-     * @var Rule[]
+     * @var array<array<int, Rule>> This is the internal map for rules to be applied
      */
-    private $rules = [];
+    private array<Rule> $rules = array();
```

Comment:
Let's do a project wide search for `array<` to review outdated documentation.

Reply:
👍 
Fixed know ones.